zerver/lib/import_realm.py

```diff
@@ -944,6 +944,22 @@ def do_import_realm(import_dir: Path, subdomain: str, processes: int = 1) -> Rea
     del data["zerver_userprofile_mirrordummy"]
     data["zerver_userprofile"].sort(key=lambda r: r["id"])
 
+    # Early versions of Zulip placed the system bots into the realms,
+    # in addition to having them as crossrealm.  Strip out any
+    # duplicates.
```

Comment:
Are we sure that's somehow the actual cause rather than something else about the way we populate `data["zerver_userprofile"]` when generating the export data causing the cross-realm bots to get pulled into that data structure? I suppose it's just not clear to me how the existence of system bots within realms in the past would cause `id`s of bots which now reside in `SYSTEM_BOT_REALM` to get pulled into that dict

The rest of the changes in this PR look good to me, just not sure about this comment.

Reply:
So I think the background is that before we invented the idea of a system bot realm, the system bots were created when creating the first ream in an organization, and thus exporting those users in the realm.

I suppose that theory would suggest an alternative fix of patching `custom_fetch_user_profile` to exclude `is_system_bot=True` and/or coordinate with `custom_fetch_user_profile_cross_realm` so as to avoid putting the duplicates in the export in the first place?